src/api/sign/nonceSignature.js

```diff
@@ -1,17 +1,16 @@
+const { makeGenerateAuthMessage } = require('dvf-utils')
 /**
- *
- * if either nonce and signature are not provided a
- * new nonce and signature are created. if nonce and
- * signature were provided the same are returned back
- *
+ * if either message and signature are not provided a
+ * new nonce/message and signature are created. if nonce
+ * and signature were provided the same are returned back
  */
-
 module.exports = async (dvf, nonce, signature) => {
   if (!(nonce && signature)) {
     nonce = Date.now() / 1000
-    signature = await dvf.sign(String(nonce).toString(16))
+    const authVersion = dvf.config.DVF.authVersion || 1
+    const message = makeGenerateAuthMessage(authVersion)(nonce)
```

Comment:
ensure endpoints that use this signature do not expect the entire message to be nonce, I remember some endpoints were checking nonce against the entire messaged in sig

Reply:
To my understanding, right now in every part of the app we expect the message to simply be the nonce as string.
The tricky part is at some point we want to start using human-readable message instead of the nonce, but we need to ensure that both the client and the server agree on what format to use, hence the usage of `dvf.config.DVF.authVersion` that doesn't exist yet but for which `1` will be the old-style signing with nonce only and `2` will be the human readable message.
So the way I plan to do this is to push that client code first where `authVersion` is `1` and then push the server changes where the server sends `authVersion` as `2` but also starts expecting the message to be something human-readable.
I think the exception to that is the nonce signing with the DTK (which I think you implemented ?) where we will still sign the nonce and expect the nonce

Reply:
indeed the DTK one expects a nonce signature, I believe it also signs the nonce as a number not a string. I see so `authVersion` is being used to determine on the server side which method of signature is used and adapt accordingly.